ufl/log.py

```diff
@@ -257,3 +257,22 @@ class UFLValueError(UFLException):
     exec("%s = ufl_logger.%s" % (foo, foo))
 
 set_level(DEPRECATE)  # noqa
+
+
+class ControlDifferentiationError(Exception):
+    "Class for the exception regarding the differentiation of an ExternalOperator with respect to a control."
+
+    def __init__(self, o, index, df):
+        from ufl.core.external_operator import ExternalOperator
+        # Checks
+        if not isinstance(o, ExternalOperator):
+            ufl_logger.error("Expecting an ExternalOperator instead of : ", o)
+
+        self.pointwise_operator = o
+        self.control = o.controls[index]
+        self.diff_control = df
+
+    def message(self):
+        "Write message and raise an exception."
+        # self._log.error(*message)
+        raise Exception("WARNING : The derivative with respect to the control was taken !!")
```

Comment:
This seems wrong. Why not just `raise ControlDifferentiationError` directly?

Reply:
For now, it is raising a warning, but subsequently the aim is not to raise an error but to catch this exception and call the appropriate mechanism to compute the derivative of the external operator with respect to its control (e.g backpropagation for neural network).